src/rpc/net.cpp

```diff
@@ -303,6 +304,23 @@ static RPCHelpMan addnode()
         throw std::runtime_error(
             self.ToString());
     }
+    const std::string manual_name{ConnectionTypeAsString(ConnectionType::MANUAL)};
+    const std::string manual_block_relay_name{ConnectionTypeAsString(ConnectionType::MANUAL_BLOCK_RELAY)};
+    if (strCommand == "remove" && !request.params[2].isNull()) {
+        throw JSONRPCError(RPC_INVALID_PARAMS, "Cannot specify a connection type with 'remove'");
+    }
+    std::string connection_type_name{manual_name};
```

Comment:
perhaps there is a way to simplify it:
```diff
diff --git a/src/rpc/net.cpp b/src/rpc/net.cpp
index 6155af323..b9e082a63 100644
--- a/src/rpc/net.cpp
+++ b/src/rpc/net.cpp
@@ -304,22 +304,19 @@ static RPCHelpMan addnode()
         throw std::runtime_error(
             self.ToString());
     }
-    const std::string manual_name{ConnectionTypeAsString(ConnectionType::MANUAL)};
-    const std::string manual_block_relay_name{ConnectionTypeAsString(ConnectionType::MANUAL_BLOCK_RELAY)};
+
     if (strCommand == "remove" && !request.params[2].isNull()) {
         throw JSONRPCError(RPC_INVALID_PARAMS, "Cannot specify a connection type with 'remove'");
     }
-    std::string connection_type_name{manual_name};
+
+    const std::string manual_block_relay_name{ConnectionTypeAsString(ConnectionType::MANUAL_BLOCK_RELAY)};
+    ConnectionType connection_type{ConnectionType::MANUAL};
     if (!request.params[2].isNull()) {
-        connection_type_name = request.params[2].get_str();
-    }
-    ConnectionType connection_type;
-    if (connection_type_name == manual_block_relay_name) {
-        connection_type = ConnectionType::MANUAL_BLOCK_RELAY;
-    } else if (connection_type_name == manual_name) {
-        connection_type = ConnectionType::MANUAL;
-    }  else {
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Unsupported connection type: " + connection_type_name);
+        const std::string connection_type_name{request.params[2].get_str()};
+        if (connection_type_name == manual_block_relay_name)
+            connection_type = ConnectionType::MANUAL_BLOCK_RELAY;
+        else
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Unsupported connection type: " + connection_type_name);
     }
 
     NodeContext& node = EnsureAnyNodeContext(request.context);
```

Reply:
Yes, thanks, that's a bit simpler and should be equivalent, so I took your suggestion!